mods/gen2/scripts.js

```diff
@@ -875,7 +875,10 @@ exports.BattleScripts = {
 			this.debug('instafaint: ' + this.faintQueue.map('target').map('name'));
 			this.faintMessages(true);
 			target.faint();
-			this.queue = [];
+			// Skip to the residuals, don't allow any further moves.
+			this.queue = this.queue.filter(function (action) {
+				return action.choice === 'residual';
+			});
```

Comment:
Wait, this is redundant with the `faintMessages` check. Why do people do these things? :|